app/models/manageiq/providers/embedded_automation_manager/configuration_script_payload.rb

```diff
@@ -1,3 +1,65 @@
 class ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScriptPayload <
   ManageIQ::Providers::AutomationManager::ConfigurationScriptPayload
+
+  include AnsiblePlaybookMixin
+
+  # create ServiceTemplate and supporting ServiceResources and ResourceActions
+  # options
+  #   :name
+  #   :description
+  #   :config_info
+  #     :variables
+  #     :hosts
+  #     :credential_id
+  #     :network_credential_id
+  #     :cloud_credential_id
+  #     :playbook_id
+  #
+  def self.run(options, auth_user = nil)
+    auth_user ||= 'system'
+    name = options[:name]
+    description = options[:description]
+    info = options[:config_info]
+    results, tower_id = ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScriptPayload.send(:create_job_template, build_name(name), description, info, auth_user)
+    launch(results, tower_id, auth_user)
+  end
+
+  def self.launch(results, tower_id, auth_user)
+    task_id = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript.launch_in_provider_queue(tower_id, results.id, results.name, auth_user)
```

Comment:
the job template `ConfigurationScript` class should have an instance method `launch_in_provider` that already knows the tower_id.